ImagesScrapper.py

```diff
@@ -30,22 +32,40 @@
                 "full body person portrait",
                 "person smiling"]
 # search_terms = ["armed masked thief"]
+
+class TimeoutException(Exception):
+    pass
+
 class timeout:
 
     def __init__(self, seconds= 1, error_message="Timeout"):
         self.seconds = seconds
         self.error_message = error_message
-
+        self.os_is_windows = platform.system().lower == 'windows'
+    
     def handle_timeout(self, signum, frame):
         raise TimeoutError(self.error_message)
 
     def __enter__(self):
-        signal.signal(signal.SIGALRM, self.handle_timeout)
-        signal.alarm(self.seconds)
+        if self.os_is_windows:
+            # For better portability a timeout class for windows is needed
+            self.timer = threading.Timer(self.seconds, self._raise_timeout)
+            self.timer.start
```

Comment:
threading.timer won't interrupt any main thread in case it's in blocking operation (i,e I/O or sleep) it waits for it to complete and raises an exception